src/erlcloud_cloudsearch.erl

```diff
@@ -0,0 +1,675 @@
+                         AnalysisScheme :: string(),
+                         DefaultValue :: string(),
+                         HighlightEnabled :: boolean(),
+                         ReturnEnabled :: boolean(),
+                         SortEnabled :: boolean(),
+                         SourceField :: string()) -> proplist().
+text_field_options(FieldName, AnalysisScheme, DefaultValue, HighlightEnabled,
+                   ReturnEnabled, SortEnabled, SourceField) ->
+    [{"IndexFieldName", FieldName},
+     {"IndexFieldType", "text"},
+     {"TextOptions.AnalysisScheme", AnalysisScheme},
+     {"TextOptions.DefaultValue", DefaultValue},
+     {"TextOptions.HighlightEnabled", HighlightEnabled},
+     {"TextOptions.ReturnEnabled", ReturnEnabled},
+     {"TextOptions.SortEnabled", SortEnabled},
+     {"TextOptions.SourceField", SourceField}].
+
+%%------------------------------------------------------------------------------
+%% Index field: TextArrayOptions
+
+%% http://docs.aws.amazon.com/cloudsearch/latest/developerguide/API_TextArrayOptions.html
+%%------------------------------------------------------------------------------
+-spec text_array_field_options(FieldName :: string()) -> proplist().
+text_array_field_options(FieldName) ->
+    [{"IndexFieldName", FieldName},
+     {"IndexFieldType", "text-array"}].
+
+-spec text_array_field_options(FieldName :: string(),
+                               AnalysisScheme :: string(),
+                               DefaultValue :: string(),
+                               HighlightEnabled :: boolean(),
+                               ReturnEnabled :: boolean(),
+                               SourceFields :: string()) -> proplist().
+text_array_field_options(FieldName, AnalysisScheme, DefaultValue,
+                         HighlightEnabled, ReturnEnabled, SourceFields) ->
+    [{"IndexFieldName", FieldName},
+     {"IndexFieldType", "text-array"},
+     {"TextArrayOptions.AnalysisScheme", AnalysisScheme},
+     {"TextArrayOptions.DefaultValue", DefaultValue},
+     {"TextArrayOptions.HighlightEnabled", HighlightEnabled},
+     {"TextArrayOptions.ReturnEnabled", ReturnEnabled},
+     {"TextArrayOptions.SourceFields", SourceFields}].
+
+%%==============================================================================
+%% Internal Functions
+%%==============================================================================
+
+cloudsearch_query(Config, Action, Params) ->
+    cloudsearch_query(Config, Action, Params, ?API_VERSION).
+
+cloudsearch_query(Config, Action, Params, ApiVersion) ->
+    QParams = [{"Action", Action}, {"Version", ApiVersion}|Params],
+    case erlcloud_aws:aws_request4(post, undefined,
+                                   Config#aws_config.cloudsearch_host,
+                                   undefined, "/", QParams, "cloudsearch",
+                                   [{"Accept", "application/json"}], Config) of
+    {ok, Response} ->
+        {ok, jsx:decode(Response)};
```

Comment:
let me revisit if returning raw results just converted to proplists is fine 
@KingBrewer i'll be back to you on this